iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -70,19 +91,125 @@ static Value transpose(mlir::Location loc, PatternRewriter &rewriter,
   return transposedOp.getResult(0);
 };
 
-// Collapses a 4d tensor operand to 2d given its target shape.
+// Collapses a 4d tensor input to 2d given its target shape.
 // Example: (M1, M0, N1, N0) -> (M, N)
 static Value collapseTo2D(mlir::Location loc, PatternRewriter &rewriter,
-                          Value operand, ArrayRef<int64_t> targetShape) {
-  auto operandType = operand.getType().cast<RankedTensorType>();
+                          Value input, ArrayRef<int64_t> targetShape) {
+  auto inputType = input.getType().cast<RankedTensorType>();
   auto targetType =
-      RankedTensorType::get(targetShape, operandType.getElementType());
+      RankedTensorType::get(targetShape, inputType.getElementType());
   SmallVector<ReassociationIndices> collapseIndices = {{0, 1}, {2, 3}};
   Value reshapedOperand = rewriter.create<linalg::TensorCollapseShapeOp>(
-      loc, targetType, operand, collapseIndices);
+      loc, targetType, input, collapseIndices);
   return reshapedOperand;
 }
 
+// Returns true if an input of the given |inputStaticShape| needs padding to
+// ensure that its shape will be a multiple of |tileShape|. That's always true
+// in the dynamic shape case.
+static bool needsPadding(ArrayRef<int64_t> inputStaticShape,
+                         ArrayRef<int64_t> tileShape) {
+  assert(inputStaticShape.size() == tileShape.size());
+  for (int i = 0; i < inputStaticShape.size(); i++) {
+    if (inputStaticShape[i] == ShapedType::kDynamicSize) {
+      return true;
+    }
+    if (inputStaticShape[i] % tileShape[i]) {
```

Comment:
I don't know what any style guide says about it, but to me, an explicit comparison here is more readable